accumulators/include/alps/accumulators/feature/binning_analysis.hpp

```diff
@@ -255,7 +255,7 @@ namespace alps {
                         using alps::numeric::check_size;
 
                         B::operator()(val);
-                        if(B::count() == (1 << m_ac_sum2.size())) {
+                        if(static_cast<size_t>(B::count()) == (1UL << m_ac_sum2.size())) {
```

Comment:
Do we need this `static_cast<size_t>`?  `count()` is supposed to return `count_type<T>::type` which is `boost::uint64_t`. Maybe it's better to redefine it in [count.hpp:31](https://github.com/ALPSCore/ALPSCore/blob/master/accumulators/include/alps/accumulators/feature/count.hpp#L31)?